applications/GeoMechanicsApplication/tests/cpp_tests/custom_utilities/test_geometry_utilities.cpp

```diff
@@ -0,0 +1,259 @@
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    PointerVector<Node> nodes;
+    nodes.push_back(Kratos::make_intrusive<Node>(1, 0.0, -7.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(2, 0.0, 10.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(3, 0.0, -7.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(4, 0.0, 10.0, 0.0));
+    const LineInterfaceGeometry<Line2D2<Node>> geometry(1, nodes);
+
+    const auto local_coordinate = array_1d<double, 3>{0.0, 0.0, 0.0};
+    Matrix     rotation_matrix =
+        GeometryUtilities::Calculate2DRotationMatrixForLineGeometry(geometry, local_coordinate);
+
+    // clang-format off
+    Matrix expected_rotation_matrix(2, 2);
+    expected_rotation_matrix <<= 0.0, -1.0,
+                                 1.0,  0.0; // Rotation of 90 degrees counterclockwise
+    // clang-format on
+    KRATOS_EXPECT_MATRIX_NEAR(expected_rotation_matrix, rotation_matrix, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeometryUtilities_UnityRotationFor3Plus3NodedInterface_WithNonUnitLength,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    PointerVector<Node> nodes;
+    nodes.push_back(Kratos::make_intrusive<Node>(1, 0.0, 0.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(2, 1.0, 0.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(3, 2.0, 0.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(4, 0.0, 0.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(5, 1.0, 0.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(6, 2.0, 0.0, 0.0));
+    const LineInterfaceGeometry<Line2D3<Node>> geometry(1, nodes);
+
+    const auto local_coordinate = array_1d<double, 3>{0.0, 0.0, 0.0};
+    Matrix     rotation_matrix =
+        GeometryUtilities::Calculate2DRotationMatrixForLineGeometry(geometry, local_coordinate);
+
+    KRATOS_EXPECT_MATRIX_NEAR(Matrix{IdentityMatrix{2}}, rotation_matrix, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeometryUtilities_UnityRotationForCurved3Plus3NodedInterface_WithNonUnitLength,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    PointerVector<Node> nodes;
+    nodes.push_back(Kratos::make_intrusive<Node>(1, 0.0, 0.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(2, 2.0, 0.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(3, 1.0, -1.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(4, 0.0, 0.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(5, 2.0, 0.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(6, 1.0, -1.0, 0.0));
+    const LineInterfaceGeometry<Line2D3<Node>> geometry(1, nodes);
+
+    const auto local_coordinate = array_1d<double, 3>{0.0, 0.0, 0.0};
+    Matrix     rotation_matrix =
+        GeometryUtilities::Calculate2DRotationMatrixForLineGeometry(geometry, local_coordinate);
+
+    KRATOS_EXPECT_MATRIX_NEAR(Matrix{IdentityMatrix{2}}, rotation_matrix, 1e-6)
```

Comment:
Just to double-check: only for the case of $\xi = 0$ we will get the identity matrix? For any other $\xi$ in the range of -1 to +1, I would expect a non-identity matrix due the geometry's curvature. Is that correct?

Reply:
Yes, that is correct, I clarified the test title